### PR TITLE
Icu 12860 retain values in form fields in credential library when switching types

### PR DESCRIPTION
**Pull request by dvonanderson**
:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/ICU-12860)

## Description
- Extract the name and description properties from the object returned by the ternary operator. If this.currentModel exists and is new, it will extract these properties from this.currentModel. Otherwise, it will try to extract them from an empty object, which will result in name and description being undefined.
- Fixed order of attributes to align with code base standard

## Screenshots (if appropriate):

https://github.com/hashicorp/boundary-ui/assets/24277002/c9aaebdd-9348-4b9c-83da-217b5634a62c

## How to Test
- Navigate to Orgs and any Project
- Navigate to Credential Stores
- Click on any Vault Store in list
- Click on the Credential Libraries tab
- Click on New Credential Library from manage dropdown
- Add a `name` and `description` and select then select the unselected type card and ensure that the data in those two fields persist

:technologist: [Admin preview](PATH_TO_FEATURE)

:desktop_computer: [Desktop preview](PATH_TO_FEATURE)

## Checklist:

- [x] I have added before and after screenshots for UI changes
~~- [ ] I have added JSON response output for API changes~~
- [x] I have added steps to reproduce and test for bug fixes in the description
~~- [ ] I have commented on my code, particularly in hard-to-understand areas~~
- [x] My changes generate no new warnings
~~- [ ] I have added tests that prove my fix is effective or that my feature works~~

### CODE REVIEW

File: ui/admin/app/routes/scopes/scope/credential-stores/credential-store/credential-libraries/new.js [hunk 1]
```diff
@@ -52,16 +52,21 @@ export default class ScopesScopeCredentialStoresCredentialStoreCredentialLibrari
...
+      : {};
 
     // Set the type to generic vault if feature flag isn't enabled in cases where
     // user sets the query parameter manually
-    if (!this.features.isEnabled('ssh-target')) {
-      type = TYPE_CREDENTIAL_LIBRARY_VAULT_GENERIC;
-    }
+    type = this.features.isEnabled('ssh-target')
+      ? type
+      : TYPE_CREDENTIAL_LIBRARY_VAULT_GENERIC;
```
Comment (DhariniJeeva) [inline ui/admin/app/routes/scopes/scope/credential-stores/credential-store/credential-libraries/new.js:63]: do we need this feature flag check at all? 

if the `type` is found in the model, set it else use the default type which is VAULT_GENERIC

Comment (DhariniJeeva) [reply ui/admin/app/routes/scopes/scope/credential-stores/credential-store/credential-libraries/new.js:63]: And we already set the default type on L51 😅  

this additional `type` reassignment seems redundant to me

Comment (dvonanderson) [reply ui/admin/app/routes/scopes/scope/credential-stores/credential-store/credential-libraries/new.js:63]: @DhariniJeeva agreed removing it

File: ui/admin/app/components/form/credential-library/vault-generic/index.hbs [hunk 2]
```diff
@@ -25,17 +24,19 @@
     {{#if @model.errors.name}}
       <F.Error as |E|>
         {{#each @model.errors.name as |error|}}
-          <E.Message>{{error.message}}</E.Message>
+          <E.Message data-test-error-message-name>
```
[the remark this answers is not in the document]
Comment (calcaide) [reply ui/admin/app/components/form/credential-library/vault-generic/index.hbs:27]: +1 to Zed message.

What is the purpose of adding a `data-test-` selector if the PR contains no test or refactor of an existing test?

**Review by calcaide — CHANGES_REQUESTED** [review]
Left comment

File: ui/admin/app/routes/scopes/scope/credential-stores/credential-store/credential-libraries/new.js (hunk 1 above)
Comment (DhariniJeeva) [reply ui/admin/app/routes/scopes/scope/credential-stores/credential-store/credential-libraries/new.js]: hmm yeah, you are right. we would need it for that use case

File: ui/admin/app/components/form/credential-library/vault-generic/index.hbs (hunk 2 above)
[the remark this answers is not in the document]
Comment (dvonanderson) [reply ui/admin/app/components/form/credential-library/vault-generic/index.hbs:27]: @calcaide thank you for that input

File: ui/admin/app/routes/scopes/scope/credential-stores/credential-store/credential-libraries/new.js [hunk 3]
```diff
@@ -52,16 +52,19 @@ export default class ScopesScopeCredentialStoresCredentialStoreCredentialLibrari
     const { id: credential_store_id } = this.modelFor(
       'scopes.scope.credential-stores.credential-store',
     );
+    const { name, description } = this.currentModel?.isNew
```
[the remark this answers is not in the document]
Comment (dvonanderson) [reply ui/admin/app/routes/scopes/scope/credential-stores/credential-store/credential-libraries/new.js:55]: @ZedLi good point, don't need that check

File: ui/admin/app/components/form/credential-library/vault-generic/index.hbs (hunk 2 above)
[the remark this answers is not in the document]
Comment (cameronperera) [reply ui/admin/app/components/form/credential-library/vault-generic/index.hbs:27]: Once we get tests added for errors, we could utilize the index from the `#each` and assign a unique `data-test` attribute. Something like:
```
{{#each @model.errors.name as |error index|}}
  <E.Message data-test-error-message-name={{index}}>
```

[the remark this answers is not in the document]
Comment (dvonanderson) [reply ui/admin/app/components/form/credential-library/vault-generic/index.hbs:27]: @cameronperera I love that idea thank you for the suggestions

[the remark this answers is not in the document]
Comment (ZedLi) [reply ui/admin/app/components/form/credential-library/vault-generic/index.hbs]: We should talk more in tech time about this if people feel differently but I don't think we should be adding index to these data test attributes. If there's nothing unique enough to identify the attribute except for the index, why are we adding an attribute in the first place? What benefit is that now adding versus just using a normal dom selector?

I personally think that's at the point of overusing these data test attributes. I think there's probably a container the attribute should be on and you can then select the children from them if needed.

[the remark this answers is not in the document]
Comment (cameronperera) [reply ui/admin/app/components/form/credential-library/vault-generic/index.hbs]: That is a fair point. To not block this PR, I am happy with it being removed and we discuss this during tech time as suggested.

[the remark this answers is not in the document]
Comment (dvonanderson) [reply ui/admin/app/components/form/credential-library/vault-generic/index.hbs]: @cameronperera already removed

File: ui/admin/app/components/form/credential-library/vault-ssh-certificate/index.hbs [hunk 4]
```diff
@@ -26,17 +26,19 @@
     {{#if @model.errors.name}}
       <F.Error as |E|>
         {{#each @model.errors.name as |error|}}
-          <E.Message>{{error.message}}</E.Message>
+          <E.Message data-test-error-message-name>
```
Comment (cameronperera) [inline ui/admin/app/components/form/credential-library/vault-ssh-certificate/index.hbs:29]: We should probably remove this `data-test` as well.

**Review by cameronperera — CHANGES_REQUESTED** [review]

File: ui/admin/app/components/form/credential-library/vault-ssh-certificate/index.hbs (hunk 4 above)
Comment (dvonanderson) [reply ui/admin/app/components/form/credential-library/vault-ssh-certificate/index.hbs:29]: @cameronperera ahh good catch

**Review by ZedLi — APPROVED** [review]

**Review by cameronperera — APPROVED** [review]

**Review by calcaide — APPROVED** [review]

**Pull request merged**